core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java

```diff
@@ -35,15 +39,21 @@
     @Autowired
     private ClientAppDataBinder clientAppDataBinder;
 
+    @Autowired
+    private AuthModuleDAO authModuleDAO;
+
     @Override
     public WAClientApp getWAClientApp(final ClientApp clientApp) {
         WAClientApp waClientApp = new WAClientApp();
         waClientApp.setClientAppTO(clientAppDataBinder.getClientAppTO(clientApp));
 
         try {
+            AuthPolicyConf authPolicyConf = null;
             if (clientApp.getAuthPolicy() != null) {
+                authPolicyConf = clientApp.getAuthPolicy().getConf();
                 waClientApp.setAuthPolicyConf((clientApp.getAuthPolicy()).getConf());
             } else if (clientApp.getRealm().getAuthPolicy() != null) {
+                authPolicyConf = clientApp.getAuthPolicy().getConf();
                 waClientApp.setAuthPolicyConf((clientApp.getRealm().getAuthPolicy()).getConf());
```

Comment:
this line must be moved right after the subsequent `}`

Reply:
ah, my bad this line has to be:
` authPolicyConf=(clientApp.getRealm().getAuthPolicy()).getConf(); `
as a sequence, authPoliceConf is needed in both condition